### PR TITLE
[NFC] Remove LinalgTransformationFilter from dispatch region formation

### PR DESCRIPTION
**Pull request by matthias-springer**
This change is NFC and simplifies the code base. `LinalgTransformationFilter` was previously used to ensure that the pattern that creates dispatch regions is not applied in certain places. This change replaces the pattern with a simpler walk over the IR. (The greedy pattern rewriter is no longer used for this purpose.)

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp [hunk 1]
```diff
@@ -842,59 +821,35 @@ static LogicalResult legalizeDispatchWorkgroupOperands(
...
-    // in mind.
-    if (!hasComputeUsesOutsideDispatch(rootOp)) return failure();
-    if (rootOp->template getParentOfType<IREE::Flow::DispatchWorkgroupsOp>()) {
-      return failure();
-    }
-
-    if (failed(transformationFilter.checkAndNotify(rewriter, rootOp))) {
-      return failure();
-    }
+static SmallVector<IREE::Flow::DispatchWorkgroupsOp> createDispatchRegions(
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp:824]: Nit: Add some documentation for this please.

**Review by MaheshRavishankar — COMMENTED** [review]
Cool! Thanks for doing this! It is possible that some of this change might lead to subtle changes that might affect performance, but I think this is a better state, and as long as there arent obvious regressions, this is definitely better!

Will stamp after benchmark runs.

**Review by benvanik — APPROVED** [review]

**Comment by MaheshRavishankar** [comment]
@matthias-springer looks like the CI failures are real....

**Comment by MaheshRavishankar** [comment]
These are some developer notes (https://github.com/iree-org/iree/tree/main/benchmarks#repro-of-tflite-model-errors) that I had to help repro these. Dont know if these are still valid. You can reach out @pzread if you need some help repro-ing this locally.

**Comment by matthias-springer** [comment]
Depends on #9914 to fix the TFLite failure.

**Comment by matthias-springer** [comment]
Also depends on #9915.

**Comment by matthias-springer** [comment]
How can I trigger another benchmark run?

**Comment by benvanik** [comment]
I thought it used to do it automatically - maybe that broke? I'd try removing the label and adding it again - or see if @pzread knows the new magic :)

**Pull request closed without merging**